src/pybind/mgr/object_format.py

```diff
@@ -0,0 +1,272 @@
+# object_format.py provides types and functions for working with
+# requested output formats such as JSON, YAML, etc.
+
+import enum
+import errno
+import json
+
+from functools import wraps
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+try:
+    from typing import Protocol  # Protocol was added in Python 3.8
+except ImportError:
+
+    class Protocol:  # type: ignore
+        pass
+
+
+import yaml
+
+from mgr_module import HandlerFuncType
+
+
+class Format(enum.Enum):
+    plain = "plain"
+    json = "json"
+    json_pretty = "json-pretty"
+    yaml = "yaml"
+    xml_pretty = "xml-pretty"
+    xml = "xml"
+
+
+def _obj_to_json(obj: Any) -> Any:
+    tj = getattr(obj, "to_json", None)
+    if tj:
+        return tj()
+    return obj
```

Comment:
Yes, I chose `to_json` as it was already somewhat common in these mgr modules I was looking at.
I do think the name is unfortunate, because as your post implies, the `to_json` function does not in fact return json. It might better be named `to_dict` or `to_json_dict` or something, but I would prefer keeping backwards compatibility.

So the intent is that `to_json` continue to return _any json serializable object_. Then the new and existing code can help the authors of mgr modules cpmvert that object into actual json. The name I intended for a function that actually returns real json is `format_json`.

Reply:
> Yes, I chose `to_json` as it was already somewhat common in these mgr modules I was looking at. I do think the name is unfortunate, because as your post implies, the `to_json` function does not in fact return json. It might better be named `to_dict` or `to_json_dict` or something, but I would prefer keeping backwards compatibility.
> 
> So the intent is that `to_json` continue to return _any json serializable object_. Then the new and existing code can help the authors of mgr modules cpmvert that object into actual json. The name I intended for a function that actually returns real json is `format_json`.

Does _backward compatibility_ here mean not breaking data serialization between mgrs during an upgrade? or basically not breaking code, UT, etc?

Reply:
Not breaking existing (presumably working) cdoe.